src/Sentry/Internal/Extensions/DisposableExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Sentry.Internal.Extensions
+{
+    internal static class DisposableExtensions
+    {
+        public static void DisposeAll(this IEnumerable<IDisposable> disposables)
+        {
+            var exceptions = new List<Exception>();
+
+            foreach (var i in disposables)
+            {
+                try
+                {
+                    i.Dispose();
+                }
+                catch (Exception ex)
+                {
+                    exceptions.Add(ex);
+                }
+            }
+
+            if (exceptions.Count > 0)
+            {
+                throw new AggregateException(exceptions);
+            }
+        }
```

Comment:
Happy path == 0 allocations. Overkill?

```suggestion
            List<Exception>? exceptions = null;

            foreach (var i in disposables)
            {
                try
                {
                    i.Dispose();
                }
                catch (Exception ex)
                {
                	exception =?? new List<Exception>();
                    exceptions.Add(ex);
                }
            }

            if (exceptions?.Count > 0)
            {
                throw new AggregateException(exceptions);
            }
        }
```